baseapp/abci.go

```diff
@@ -540,6 +540,15 @@ func gRPCErrorToSDKError(err error) error {
 // createQueryContext creates a new sdk.Context for a query, taking as args
 // the block height and whether the query needs a proof or not.
 func (app *BaseApp) createQueryContext(height int64, prove bool) (sdk.Context, error) {
+	if height < 0 {
```

Comment:
why don't we change a `height` type to `uint64`?

Reply:
I tried that once, and ended up in TM https://github.com/tendermint/tendermint/blob/53022220f6699874f3ef5375d7e98d3b346cc4b0/proto/tendermint/blockchain/types.proto#L8-L11, then gave up to avoid too much casting in the sdk.

Reply:
BTW: They are planning to change it in Tendermint: https://discord.com/channels/669268347736686612/669283915743232011/770955223497179136

Reply:
I still think it's better to use `uint64` whenever we parse such data, and then cast it to `int64` for Tendermint when needed. With that we catch the errors during the request parsing -> which is much better.

Reply:
@robert-zaremba I had the same idea, but 

> then gave up to avoid too much casting in the sdk

By experience, it decreases readability and increases confusion where to cast and where not. At least right now, we know height is everywhere int64 in the SDK.

Anyways, as Emmanuel said, it's out of scope for this PR.

Reply:
I've actually seen places where we've started to use `uint64` and would prefer that we change sooner rather than later. Maybe out of scope idk... What's the timeline on tendermint making this change @marbar3778 ?

Reply:
There is the chance it lands in 0.35 (min 6 months). We haven't slated it for a specific release but it is on our list of things to do.